### PR TITLE
fix(log): Add request headers to access denied exception handler log

### PR DESCRIPTION
**Pull request by jonsie**

### CODE REVIEW

File: fiat-api/src/main/java/com/netflix/spinnaker/fiat/shared/FiatAccessDeniedExceptionHandler.java [hunk 1]
```diff
@@ -73,6 +78,18 @@ private String authorizationFailureMessage(
     return sj.toString();
   }
 
+  private Map<String, String> requestHeaders(HttpServletRequest request) {
+    Map<String, String> headers = new HashMap<>();
+
+    while (request.getHeaderNames() != null && request.getHeaderNames().hasMoreElements()) {
+      String headerKey = request.getHeaderNames().nextElement();
```
Comment (marchello2000) [inline fiat-api/src/main/java/com/netflix/spinnaker/fiat/shared/FiatAccessDeniedExceptionHandler.java]: won't the enumeration be reset every time you call `getHeaderNames()` (or give you a new enumeration in reset state?)

Comment (jonsie) [reply fiat-api/src/main/java/com/netflix/spinnaker/fiat/shared/FiatAccessDeniedExceptionHandler.java]: Yes, dumb.

File: fiat-api/src/main/java/com/netflix/spinnaker/fiat/shared/FiatAccessDeniedExceptionHandler.java [hunk 2]
```diff
@@ -73,6 +79,20 @@ private String authorizationFailureMessage(
     return sj.toString();
   }
 
+  private Map<String, String> requestHeaders(HttpServletRequest request) {
+    Map<String, String> headers = new HashMap<>();
+
+    if (request.getHeaderNames() != null) {
+      for (Enumeration<?> e = request.getHeaderNames(); e.hasMoreElements(); ) {
```
Comment (marchello2000) [inline fiat-api/src/main/java/com/netflix/spinnaker/fiat/shared/FiatAccessDeniedExceptionHandler.java]: nit: `e` -> `headerNames`

Comment (jonsie) [reply fiat-api/src/main/java/com/netflix/spinnaker/fiat/shared/FiatAccessDeniedExceptionHandler.java]: I actually just changed that to `h`

**Pull request merged**